### ISSUE TITLE
Griddle delayed load of data without useExternal

### ISSUE THREAD

**Issue by jpierson** [GriddleGriddle/Griddle#303]
I have a case where I have a deferred loading of data but then after that point I want Griddle to handle all of the sorting and paging and such. If I load Griddle with an empty array for the **results** then after the data made available and Griddle is re-rendered the grid with the page count shows up but no data and no columns show. I'm assuming that this has to do with Griddle expecting some data with the expected shape upon first render when not using external data. The problem with the **useExternal** feature however is that it appears that once you use external data that there is no way to tell Griddle to handle the paging and sorting after the data is initially loaded.

Is there a way to achieve what I need without providing all of the sorting and paging logic for my single delay load of data? My workaround for now is to surround Griddle with a check such as shown below.

``` jsx
{isDataLoaded && <Griddle results={data} columnMetadata={myColumnsMetadata} />}
```

I'm assuming that this can be demonstrated with a simple setTimeout delay but I haven't had a chance to put together a reproduction case yet.

**Comment by legomind** [comment]
I load my data into a redux store, setting `allMembers.pending` to `true` while waiting for the data, and back to `false` once loaded into `allMembers.payload`.

That lets me show a nice spinner while the data is loading.

```js
export default handleActions({
  [GET_ALL_MEMBERS_PENDING]: state => ({
    ...state,
    pending: true,
  }),

  [GET_ALL_MEMBERS_REJECTED]: { throw: (state, action) => ({
    ...state,
    pending: false,
    error: action.payload,
  }) },

  [GET_ALL_MEMBERS_FULFILLED]: (state, action) => ({
    ...state,
    pending: false,
    payload: action.payload,
  }),
}, initialState)

```

```jsx
{ allMembers.pending
            ? <Loader color='white' />

            : <Griddle results={ allMembers.payload } /> }
```

**Comment by jpierson** [comment]
@legomind Ok, so it sounds like you are confirming that I'm not alone in resorting to this technique to handle at least what appears to me to be a deficiency in Griddle handling initially absent results. I still think it would be useful to take advantage of the same loading component concepts as is available when using external data.

**Comment by ryanlanciaux** [comment]
@legomind Thanks for the example of the code.

Sorry that this is not something that's working in Griddle right now. If someone would want to submit a PR (with tests please) that encapsulates this logic that would be incredible.

**Comment by legomind** [comment]
@ryanlanciaux Should I work from master? I am inclined to work on v1 alpha, even though its not really ready just yet.

**Comment by ryanlanciaux** [comment]
Based off of master would be best for this -- the codebase for 1.0 will need to handle it in a different way (I would love help on that though too at some point if any availability :smile:)

**Comment by ryanlanciaux** [comment]
Sorry about the really long delay here. We have recently shipped Griddle 1.0 and this should no longer be an issue. Please feel free to open a new issue if there are any problems that you encounter. :+1:
